components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/web/cache-mediator/update-mediator.jsp

```diff
@@ -15,83 +15,89 @@
-    String cacheScope = request.getParameter("cacheScope");
+    String cacheId = request.getParameter("cacheId");
     String cacheType = request.getParameter("cacheType");
-    String cacheTimeout = request.getParameter("cacheTimeout");
-    String maxMsgSize = request.getParameter("maxMsgSize");
-    String impType = request.getParameter("impType");
-    String maxSize = request.getParameter("maxSize");
+    String protocolType = request.getParameter("protocolType").trim();
+    String cacheTimeout = request.getParameter("cacheTimeout").trim();
+    String maxMsgSize = request.getParameter("maxMsgSize").trim();
+    String methods = request.getParameter("methods");
+    String headersToExclude = request.getParameter("headersToExclude");
+    String responseCodes = request.getParameter("responseCodes");
+    String hashGen = request.getParameter("hashGen");
+    String maxSize = request.getParameter("maxSize").trim();
     String sequenceOption = request.getParameter("sequenceOption");
 
-    if(notNullChecker(cacheid)){
-        cacheMediator.setId(cacheid);
-    }
-    if(notNullChecker(cacheScope)){
-        cacheMediator.setScope(cacheScope.toLowerCase());
-    }
-    if(notNullChecker(cacheType)){
-        if(cacheType.equalsIgnoreCase("Collector")){
+    if (notNullChecker(cacheType)) {
+        if (cacheType.equalsIgnoreCase("Collector")) {
             cacheMediator.setCollector(true);
-        }
-        else{
+        } else {
             cacheMediator.setCollector(false);
         }
     }
-    if(notNullChecker(cacheTimeout)){
-        try{
-        cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
-        }
-        catch(NumberFormatException e){
+    if (notNullChecker(cacheTimeout)) {
+        try {
+            cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
+        } catch (NumberFormatException e) {
 
         }
     }
-    if(notNullChecker(maxMsgSize)){
-        try{
-        cacheMediator.setMaxMessageSize(Integer.parseInt(maxMsgSize));
-        }
-        catch(NumberFormatException e){
+    if (notNullChecker(maxMsgSize)) {
+        try {
+            cacheMediator.setMaxMessageSize(Integer.parseInt(maxMsgSize));
+        } catch (NumberFormatException e) {
 
         }
     }
-    if(notNullChecker(impType)){
+
+    if (notNullChecker(protocolType)) {
+        cacheMediator.setProtocolType(protocolType);
+    }
+
+    if (notNullChecker(methods)) {
+        cacheMediator.setHTTPMethodsToCache(methods);
+    }
+
+    if (notNullChecker(headersToExclude)) {
+        cacheMediator.setHeadersToExcludeInHash(headersToExclude);
+    }
+
+    if (notNullChecker(responseCodes)) {
+        cacheMediator.setResponseCodes(responseCodes);
+    }
+
+    if (notNullChecker(hashGen)) {
+        cacheMediator.setDigestGenerator(hashGen);
+    }
+
+    if (notNullChecker(maxSize)) {
         try {
             cacheMediator.setInMemoryCacheSize(Integer.parseInt(maxSize));
         } catch (NumberFormatException e) {
```

Comment:
This is for the UI. The check to make sure it is integer is carried out in the mediator-util.js so the exception is ignored here.